lib/ansible/modules/windows/win_package.ps1

```diff
@@ -1,1329 +1,524 @@
-PackageConfigurationComplete=Package configuration finished
-PackageConfigurationStarting=Package configuration starting
-InstalledPackage=Installed package
-UninstalledPackage=Uninstalled package
-NoChangeRequired=Package found in desired state, no action required
-RemoveExistingLogFile=Remove existing log file
-CreateLogFile=Create log file
-MountSharePath=Mount share to get media
-DownloadHTTPFile=Download the media over HTTP or HTTPS
-StartingProcessMessage=Starting process {0} with arguments {1}
-RemoveDownloadedFile=Remove the downloaded file
-PackageInstalled=Package has been installed
-PackageUninstalled=Package has been uninstalled
-MachineRequiresReboot=The machine requires a reboot
-PackageDoesNotAppearInstalled=The package {0} is not installed
-PackageAppearsInstalled=The package {0} is already installed
-PostValidationError=Package from {0} was installed, but the specified ProductId and/or Name does not match package details
-'@
-}
+$ErrorActionPreference = 'Stop'
 
-$Debug = $true
-Function Trace-Message
-{
-    param([string] $Message)
-    if($Debug)
-    {
-        Write-Verbose $Message
-    }
-}
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
-$CacheLocation = "$env:ProgramData\Microsoft\Windows\PowerShell\Configuration\BuiltinProvCache\MSFT_PackageResource"
+$arguments = Get-AnsibleParam -obj $params -name "arguments" -type "str"
+$expected_return_code = Get-AnsibleParam -obj $params -name "expected_return_code" -type "list" -default @(0, 3010)
+$name = Get-AnsibleParam -obj $params -name "name" -type "str"
+$path = Get-AnsibleParam -obj $params -name "path" -type "str"
+$product_id = Get-AnsibleParam -obj $params -name "product_id" -type "str" -aliases "productid"
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "absent","present" -aliases "ensure"
+$user_name = Get-AnsibleParam -obj $params -name "user_name" -type "str"
+$user_password = Get-AnsibleParam -obj $params -name "user_password" -type "str" -failifempty ($user_name -ne $null)
+$validate_certs = Get-AnsibleParam -obj $params -name "validate_certs" -type "bool" -default $true
+$creates_path = Get-AnsibleParam -obj $params -name "creates_path" -type "path"
+$creates_version = Get-AnsibleParam -obj $params -name "creates_version" -type "str"
+$creates_service = Get-AnsibleParam -obj $params -name "creates_service" -type "str"
 
-Function Throw-InvalidArgumentException
-{
-    param(
-        [string] $Message,
-        [string] $ParamName
-    )
-    
-    $exception = new-object System.ArgumentException $Message,$ParamName
-    $errorRecord = New-Object System.Management.Automation.ErrorRecord $exception,$ParamName,"InvalidArgument",$null
-    throw $errorRecord
+$result = @{
+    changed = $false
+    reboot_required = $false
```

Comment:
I'll put it in there but will document that we will switch to `reboot_required` in a future release.